backend/src/controllers/useCommonComponents/index.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * Get a specific item by ID.
+ *
+ * @param   {FastifyRequest} request FastifyRequest is an instance of the standard http or http2 request objects.
+ * @param   {FastifyReply}   reply   FastifyReply is an instance of the standard http or http2 reply types.
+ * @returns {object}
+ */
+const { config, chesApi, cdogsApi } = require("../../facilities/bcgov_cc_token");
+
+const { ClientCredentials } = require("simple-oauth2");
+const axios = require("axios");
+
+const useCommonComponents = (healthApi, instance) => {
+  const getHealth = async (request, reply, api) => {
+    // Gets grant type client credentials.
+    const client = new ClientCredentials(config);
+    const tokenParams = { scope: "<scope>" };
+
+    const accessToken = await client.getToken(tokenParams);
+    // Using Axios to call api endpoint with Bearer token.
+    const axiosInstance = axios.create({
+      baseURL: `${"cdogs" === instance ? cdogsApi : chesApi}${healthApi}`,
```

Comment:
I think it would be nice to create separate CDOGS and CHES clients since their functionality will be pretty different once we actually start implementing them beyond just the health checks. They can share the config and getToken request but then the rest of the clients I think make sense being in separate files.

Reply:
We are doing this like the controllers, if they share similar API functionality, it will go here.  We shouldn't duplicate code and although this file only has one function in common between the two, it still removes duplicate code.

Reply:
Like the CHES client will need a "sendEmail()" function or whatever and the CDOGS' will need a "getPdf()" (I don't know what CDOGS does). So I'm thinking you'll want to create a client for each and then controllers/cdogs would make its calls through the CDOGS client and controllers/ches through the CHES client. Since they both do the same thing right now there's no difference but there will be later.

Reply:
i think this is what the ches and cdogs controller is for.  It looks like @ASpiteri-BCGov is using this common component for similarities in the api, including getting token
though not sure what is being done with the base url with the healthapi